src/scratchOrgSettingsGenerator.ts

```diff
@@ -119,11 +118,30 @@ export default class SettingsGenerator {
       status: result.status,
     });
 
-    while (!breakPooling.includes(result.status)) {
+    const timer = (timeout?: number) => {
+      let expired = false;
+      setTimeout(() => {
+        expired = true;
+      }, timeout);
+      return (): boolean => expired;
+    };
+
+    const sleep = async (delay?: number): Promise<void> => {
```

Comment:
if you decide to do it this way, there's already a `sleep` in `kit` that you can use instead of writing another one.